cmake/external/helper_functions.cmake

```diff
@@ -163,19 +163,32 @@ macro(onnxruntime_fetchcontent_makeavailable)
           get_property(subdir_import_targets DIRECTORY "${__cmake_srcdir}" PROPERTY BUILDSYSTEM_TARGETS)
           foreach(subdir_target ${subdir_import_targets})
             if(TARGET ${subdir_target})
-              get_target_property(subdir_target_type ${subdir_target} TYPE)            
+              get_target_property(subdir_target_type ${subdir_target} TYPE)
               if(subdir_target_type STREQUAL "EXECUTABLE")
                 get_target_property(subdir_target_osx_arch ${subdir_target} OSX_ARCHITECTURES)
                 if (subdir_target_osx_arch)
-                  if (NOT ${CMAKE_HOST_SYSTEM_PROCESSOR} IN_LIST subdir_target_osx_arch)                              
+                  if (NOT ${CMAKE_HOST_SYSTEM_PROCESSOR} IN_LIST subdir_target_osx_arch)
                     message("Added an executable target ${subdir_target} but it can not run natively on ${CMAKE_HOST_SYSTEM_PROCESSOR}, we will try to modify it")
                   endif()
                 endif()
               endif()
               set_target_properties(${subdir_target} PROPERTIES FOLDER "External")
               set_target_properties(${subdir_target} PROPERTIES COMPILE_WARNING_AS_ERROR OFF)
             endif()
-          endforeach()          
+          endforeach()
+          if(MSVC)
+          # Replace /Zi and /ZI with /Z7
+          if(MSVC_Z7_OVERRIDE)
```

Comment:
You may try this cmake variable: https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_DEBUG_INFORMATION_FORMAT.html

Reply:
OK, let's upgrade cmake to 3.25 first

Reply:
I tried, but it doesn't work so far